integration/spark/src/test/java/io/openlineage/spark/agent/lifecycle/StaticExecutionContextFactory.java

```diff
@@ -44,7 +44,7 @@ public StaticExecutionContextFactory(OpenLineageContext sparkContext) {
    * @throws InterruptedException
    */
   public static void waitForExecutionEnd() throws InterruptedException, TimeoutException {
-    boolean acquired = semaphore.tryAcquire(1, TimeUnit.SECONDS);
+    boolean acquired = semaphore.tryAcquire(5, TimeUnit.SECONDS);
```

Comment:
So, as I got the problem is with concurrent execution of tests with SparkSession, cause it should be only one, right?

Maybe we can configure gradle to run test classes in sync and fork only test classes instead of each test?

Reply:
No, the problem is not with Gradle's concurrent execution of the tests. It is that the Spark listener always runs in a separate thread from the main Spark execution- this is the case in production as well as in tests. Since they run in separate threads, the execution of, e.g., `DataFrame.write()` may return before the listener's `onJobEnd` method completes. We have to wait for the listener's code to finish executing before we test that the client calls were made, so we use the semaphore as a way to validate that the listener code is done.